File: frontend_dbms/src/app/shared/models/readingprogress.ts
import {User} from "./user";
import {Book} from "./book";

export interface ReadingProgress {
  id: number;
  // user: User;
  userId: number;
  bookId: number;
  progress: 'NOT_STARTED' | 'IN_PROGRESS' | 'COMPLETED';
  bookImageUrl: string;
  bookTitle: string;
}

File: backend-foreveread/src/main/java/dbmsforeveread/foreveread/readingprogress/ReadingProgress.java
package dbmsforeveread.foreveread.readingprogress;

import dbmsforeveread.foreveread.book.Book;
import dbmsforeveread.foreveread.user.User;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class ReadingProgress {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "book_id")
    private Book book;

    @Enumerated(EnumType.STRING)
    private ReadingStatus progress;
}
